jbehave-core/src/main/java/org/jbehave/core/parsers/JsonAliasParser.java

```diff
@@ -0,0 +1,164 @@
+package org.jbehave.core.parsers;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.Validate.isTrue;
+
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import org.jbehave.core.configuration.Keywords;
+import org.jbehave.core.model.Alias;
+import org.jbehave.core.model.AliasVariant;
+import org.jbehave.core.steps.StepType;
+
+public class JsonAliasParser implements AliasParser {
+
+    private final Keywords keywords;
+
+    public JsonAliasParser(Keywords keywords) {
+        this.keywords = keywords;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Collection<Alias> parse(Set<String> aliasesAsStrings) {
+        if (aliasesAsStrings.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Gson gson = new Gson();
+        Type aliasType = new TypeToken<List<AliasContainer>>() {}.getType();
+
+        return aliasesAsStrings.stream()
+                               .map(json -> (List<AliasContainer>) gson.fromJson(json, aliasType))
+                               .flatMap(List::stream)
+                               .map(validate())
+                               .collect(groupingBy(AliasContainer::getName, collectingAndThen(toList(),
+                                       mergeAliases())))
+                               .values()
+                               .stream()
+                               .map(this::convert)
```

Comment:
does it makes sense to follow same approach? either validate() convert() vs this::validate, this::convert?